From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Miles Chen <miles.chen@mediatek.com>
Date: Wed, 27 Jan 2021 12:31:23 +0800
Subject: ANDROID: GKI: select DMADEVICES if GKI_HIDDEN_SND_SOC_CONFIGS is set

DMA_VIRTUAL_CHANNELS depends on DMADEVICES and DMA_VIRTUAL_CHANNELS
is selected by GKI_HIDDEN_SND_SOC_CONFIGS.

To fix unmet dependency in x86_64, select DMADEVICES under
GKI_HIDDEN_SND_SOC_CONFIGS.

Also, drop CONFIG_DMADEVICES=y in arch/arm64/configs/gki_defconfig.

WARNING: unmet direct dependencies detected for DMA_VIRTUAL_CHANNELS
  Depends on [n]: DMADEVICES [=n]
    Selected by [y]:
      - GKI_HIDDEN_SND_SOC_CONFIGS [=y]

Bug: 174628645
Signed-off-by: Miles Chen <miles.chen@mediatek.com>
Change-Id: I1de491c10542f1b7c9c8754c65415b028d3bc922
[joneslee: Removed defconfig elements]
Signed-off-by: Lee Jones <lee.jones@linaro.org>
---
 init/Kconfig.gki | 1 +
 1 file changed, 1 insertion(+)

diff --git a/init/Kconfig.gki b/init/Kconfig.gki
--- a/init/Kconfig.gki
+++ b/init/Kconfig.gki
@@ -48,6 +48,7 @@ config GKI_HIDDEN_SND_SOC_CONFIGS
 	select SND_SOC_GENERIC_DMAENGINE_PCM if (SND_SOC && SND)
 	select SND_PCM_IEC958
 	select SND_SOC_COMPRESS if (SND_SOC && SND)
+	select DMADEVICES
 	select DMA_VIRTUAL_CHANNELS
 	help
 	  Dummy config option used to enable hidden SND_SOC configs.
